### PR TITLE
misc(timing): move timing-trace to async events

### PR DESCRIPTION
**Pull request by paulirish**
Motivated by #6410

I was avoiding async events for .. mysterious reasons before. But really these will be overlapping, so it makes more sense to be async.

Overlapping happens in promise.all()s often:
image

Here's how a few timing traces look now:
image
image

### CODE REVIEW

File: lighthouse-core/lib/timing-trace-saver.js [hunk 1]
```diff
@@ -5,70 +5,55 @@
...
-
+  entries.forEach(entry => {
     /** @type {LH.TraceEvent} */
-    const traceEvent = {
-      name: entry.name,
-      cat: entry.entryType,
+    const startEvt = {
+      // 1) Remove 'lh:' for readability
+      // 2) Colons in user_timing names get special handling in traceviewer we don't want. https://goo.gl/m23Vz7
+      //    Replace with a 'Modifier letter colon' ;)
```
Comment (patrickhulce) [inline lighthouse-core/lib/timing-trace-saver.js]: 😆

**Comment by paulirish** [comment]
thoughts?


-----------

I've added support for DevTools Perf panel (via adding a TracingStartedInBrowser evt):
image



Though note: -G/-A traces are shown in two diff tracks in chrome:tracing but that's not supported in DevTools:
image

File: lighthouse-core/test/lib/timing-trace-saver-test.js [hunk 2]
```diff
@@ -37,92 +36,11 @@ const mockEntries = [{
...
-      entryType: 'measure',
-    },
-    ];
-
-    generateTraceEvents(overlappingEntries);
-    expect(consoleError).toHaveBeenCalled();
-    expect(consoleError.mock.calls[0][0]).toContain('measures overlap');
+  it('generates a pair of trace events', () => {
+    const events = generateTraceEvents([mockEntries[0]]);
+    expect(events.slice(0, 1)).toMatchSnapshot();
```
Comment (patrickhulce) [inline lighthouse-core/test/lib/timing-trace-saver-test.js]: seems a bit odd that we call the test `generates a pair` but then only look at 1 event, just do 'em both?

File: lighthouse-core/test/lib/timing-trace-saver-test.js [hunk 3]
```diff
@@ -37,92 +36,11 @@ const mockEntries = [{
...
   });
 });
```
Comment (exterkamp) [inline lighthouse-core/test/lib/timing-trace-saver-test.js]: Add a test to hit a zero duration event?

```javascript
  it('Handles zero duration events', () => {
    const overlappingEntries = [
      {
        startTime: 10,
        name: 'zeroLength',
        duration: 0,
        entryType: 'measure',
      },
    ];
    const expected = {
      name: 'zeroLength',
      cat: 'measure',
      ts: 10000,
      dur: 0,
      args: {},
      pid: 0,
      tid: 50,
      ph: 'n',
      s: 't',
      id: '0x0',
    };
    const event = generateTraceEvents(overlappingEntries);
    assert.deepStrictEqual(event[0], expected);
  });
```

File: lighthouse-core/test/lib/timing-trace-saver-test.js (hunk 2 above)
Comment (paulirish) [reply lighthouse-core/test/lib/timing-trace-saver-test.js]: lol thx

File: lighthouse-core/test/lib/timing-trace-saver-test.js (hunk 3 above)
Comment (paulirish) [reply lighthouse-core/test/lib/timing-trace-saver-test.js]: good call. actually we dont need the `ph:'n'` now that we're on async events. 
problem solved. :)

File: lighthouse-core/lib/timing-trace-saver.js [hunk 4]
```diff
@@ -5,70 +5,48 @@
...
-    if (isOverlap) {
-      // eslint-disable-next-line no-console
-      console.error(`Two measures overlap! ${prevEntry.name} & ${entry.name}`);
-    }
-  }
-}
-
 /**
  * Generates a chromium trace file from user timing measures
  * Adapted from https://github.com/tdresser/performance-observer-tracing
```
Comment (brendankenny) [inline lighthouse-core/lib/timing-trace-saver.js]: maybe add a note about how `threadId` is used?

File: lighthouse-core/lib/timing-trace-saver.js [hunk 5]
```diff
@@ -5,70 +5,48 @@
...
-
     /** @type {LH.TraceEvent} */
-    const traceEvent = {
-      name: entry.name,
-      cat: entry.entryType,
+    const startEvt = {
+      // 1) Remove 'lh:' for readability
+      // 2) Colons in user_timing names get special handling in traceviewer we don't want. https://goo.gl/m23Vz7
+      //    Replace with a 'Modifier letter colon' ;)
+      name: entry.name.replace('lh:', '').replace(/:/g, '꞉'),
```
Comment (brendankenny) [inline lighthouse-core/lib/timing-trace-saver.js]: 😆 

maybe do an explicit `.replace(/:/g, '\ua789'),` just to make it doubly clear that the character is different?

File: lighthouse-core/lib/timing-trace-saver.js [hunk 6]
```diff
@@ -5,70 +5,48 @@
...
+      cat: 'blink.user_timing',
```
Comment (brendankenny) [inline lighthouse-core/lib/timing-trace-saver.js]: it feels like doing this should be possible without faking out a browser trace but w/e traceviewer :)

File: lighthouse-core/scripts/generate-timing-trace.js [hunk 7]
```diff
@@ -50,7 +50,7 @@ function saveTraceFromCLI() {
   // eslint-disable-next-line no-console
   console.log(`
   > Timing trace file saved to: ${traceFilePath}
-  > Open this file in chrome://tracing
+  > Open this file in DevTools perf panel (or if it's a -G/A run, use chrome://tracing)
```
Comment (brendankenny) [inline lighthouse-core/scripts/generate-timing-trace.js]: does `-G/A run` refer to a "`-G` or an `-A` run", or a `-GA` run or a regular run (that did both gathering and auditing)?

**Comment by brendankenny** [comment]
(with nits :)

File: lighthouse-core/lib/timing-trace-saver.js (hunk 6 above)
Comment (paulirish) [reply lighthouse-core/lib/timing-trace-saver.js]: traceviewer wont group these but give each measure its own track otherwise
and devtools needs it to view them.
